### PR TITLE
[DEV APPROVED] activate survive january campaign on a certain date

### PR DESCRIPTION
**Pull request by Tr4pSt3R**
[Review on Reviewable](https://reviewable.io/reviews/moneyadviceservice/frontend/1340)

### CODE REVIEW

File: app/views/articles/show.html.erb [hunk 1]
```diff
@@ -63,9 +65,10 @@
 <div class="l-nav l-article-3col-right">
   <%= render 'shared/want_to_talk' %>
   <div class='related-links--desktop' data-dough-component="StickyColumn">
-
-    <% unless @campaign_promo_excluded %>
-      <%= render 'articles/survive_january_promo' %>
+    <% if display_campaign_promo? %>
```
Comment (Tr4pSt3R) [inline app/views/articles/show.html.erb]: Wrapping a condition around another condition may not be ideal, nonetheless this is temporary and due to be destroyed by the end of January

**Comment by rnalexander** [comment]
:+1:

File: app/views/articles/show.html.erb (hunk 1 above)
Comment (Tr4pSt3R) [inline app/views/articles/show.html.erb]: @asmega any thougths on this please?

File: app/helpers/survive_january.rb [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+  GO_LIVE_MONTH = 12
+  GO_LIVE_DAY   = 28
+
+  class Campaign
+    def initialize
+      @go_live_date = Date.new(GO_LIVE_YEAR, GO_LIVE_MONTH, GO_LIVE_DAY)
+    end
+
+    def go_live?
+      Date.today == @go_live_date || Date.today > @go_live_date
```
Comment (asmega) [inline app/helpers/survive_january.rb]: cant this be a greater than or equal to?

Comment (Tr4pSt3R) [inline app/helpers/survive_january.rb]: yeah it can be. I split it up for the sake of readability

Comment (asmega) [inline app/helpers/survive_january.rb]: ok. although i don't agree it is more readable though

File: app/views/articles/show.html.erb (hunk 1 above)
Comment (asmega) [inline app/views/articles/show.html.erb]: i don't see any better alternatives

File: app/helpers/survive_january.rb (hunk 2 above)
Comment (mconnell) [inline app/helpers/survive_january.rb]: If I was fussy I'd maybe add parenthesis to make the logic a bit more obvious:
`Date.today == @go_live_date || (Date.today > @go_live_date)`
or maybe just simply the line to:
`Date.today >= @go_live_date`

¯\_(ツ)_/¯

**Comment by asmega** [comment]
:+1:

**Comment by mconnell** [comment]
:+1:

**Pull request merged**